src/pages/index.tsx

```diff
@@ -28,15 +28,43 @@ const HomePage: NextPageWithLayout = () => {
   const [componentProps, setComponentProps] = useState<Record<string, unknown>>({});
   const [showNotificationModalState, setShowNotificationModalState] = useState(false);
   const accountId = useAuthStore((store) => store.accountId);
+  const [tosData, setTosData] = useState<TosData | null>(null);
+  const cacheTosData = useMemo(() => tosData, [tosData?.latestTosVersion]);
+
+  const handleModalCloseOnEsc = useCallback(() => {
```

Comment:
This and all other functions with `useCallback` is needed to prevent flashing and unnecessary re-rendering of NotificationAlert component